### DISCUSSION TITLE
Trust algorithm

### DISCUSSION THREAD

**Discussion by hellmanhellman** [stackernews/stacker.news#615] [General]
@huumn @ekzyis 

I've been looking at the personalized view stuff (which i find a little scary) and came across the confidence(s,t,z) function and trust (random walk) algorithm and have questions. First of all, am I understanding it correctly if I say that you basically 
1. Find all the items a pair of users (for all users) have both zapped. 
2. Pick out the times they agreed (so both downzapped or neither did), separating out the first zapper ("before") from the second, and counting 10* the amount of times they did not agree.
3. Use the "before" count (user a) agreements minus 10*disagreements as the "successful agreements between the users", together with the total trials (all b:s votes except those that are tainted by coming after a), in order to generate the [predicted _p_ probability](https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Wilson_score_interval) that these users will agree again?

I might be completely off here, just wondering. And I understand if you don't want to discuss it, but I was thinking, would it not be better to use a "PageRank" type algorithm? E.g:
https://en.wikipedia.org/wiki/EigenTrust

I'm suggesting it in case you haven't heard of it (its more a computer science than a physics thing for sure) but it seems ideal for this application. It's around O(log(n)) instead of O(n^3) at least. Let me know if you want to discuss it anyway.

I vaguely know the motivation of wanting to present "truth faster", i guess thats why the personalized view stuff is going on. [It looks like k00b (user 616) also has "veto" trust power](https://github.com/stackernews/stacker.news/blob/8f590425dc601558ee8986e06c5859763a0806be/prisma/migrations/20231013002652_pwot_views/migration.sql#L40C4-L40C4), (aka moderation power) which obviously is necessary now that you host your own images. I don't have much question about that except to check if that's about right. 

With lightning, it would be so cool to have a website where you could be presented truth, instead of filtered truth. Maybe you're breedlove fans like me, and think money is truth / energy. Can't cheat. If you're interested in discussing it im here!

**Comment by huumn** [comment]
We only use `confidence` to weight trust linking ... which we do because we aren't solving the exact same problem google is solving.

> It's around O(log(n)) instead of O(n^3) at least.

You an I discussed this before when you first joined the site.

> [It looks like k00b (user 616) also has "veto" trust power](https://github.com/stackernews/stacker.news/blob/8f590425dc601558ee8986e06c5859763a0806be/prisma/migrations/20231013002652_pwot_views/migration.sql#L40C4-L40C4), (aka moderation power)

This isn't showing veto power or moderation power. It's establishing a default view for people without much zapping history. My PoV is established the same way everyone else's is.

**Reply by hellmanhellman** [reply]
I guess we did, I didn't see the troubles then, I'm grateful to have you respond. The problem is then that I don't understand the problem you are trying to solve, as I thought it was to show people relevant truth fast. Which is exactly what Google solved. 

Can I ask, what problem are you trying to solve? As it stands SN seems to be trying to become echo chamber dot Com if you get my meaning. Optimizing for "seeing more of those I'm likely to agree with" instead of Optimizing for "seeing more of those I undervalue".
Im on mobile so can't check if that's how things are done, but if they are then I honestly don't judge, you do what you need to make money. For real. A lot of people don't understand crypto so I hope you do feel some inclination to not make an echo chamber that might end up hurting a lot of non "maximalist" normal people.

So what is the site about?

**Comment by huumn** [comment]
> Can I ask, what problem are you trying to solve? 

With this algorithm? Giving people more of what they've paid for. Does that mean they'll get truth faster in some absolute sense? It depends on whether they care for absolute truth enough to pay for it. We only pay for lies we want to believe are truth. At the very least I'm making it expensive to lie to yourself.

------------

In our conversations, you seem to assume that I think I know something, that I have a grand theory. I don't. I'm making all of this up. It's all an experiment. I'm just iterating on guesses.

**Reply by hellmanhellman** [reply]
To be honest, I thought you huumn were Dr Calle on Twitter, a physicist, I dint realize I was speaking to the king. (Im Swedish and that is a respectful word). Who knows. 

Yes I think you know something about truth fast that I ain't understand. But we can go beyond that. 

The problem I'm trying to solve is the same you're working on: How do we architect a building such that its inhabitants feel exactly like a family, even though they’ve never met?


What is more truthful than the admonishing little sister? 

Honesty and Anonymity combined. How? We need both. Honesty of course. No problem! Maybe not for me and you I suppose but the woman in Beijing, she needs anonymity to be honest.

I wrestle with these problems

**Reply by huumn** [reply]
> The problem I'm trying to solve is the same you're working on: How do we architect a building such that its inhabitants feel exactly like a family, even though they’ve never met?

It's an interesting problem, but as a scientist you'll know you have to define exactly what "feeling like a family" means before you can begin solving the problem.

I'm perhaps one of the worst people to ask what "feeling like a family" means. On the spectrum of an orphan <-to-> a person raised by the best family ever, I'm much much closer to an orphan. Family is mostly an idea to me rather than an experience.

You imply identity verification might be a solution to the problem were it not for needing anonymity. At the root of this is proof of humanness. You'll want to look for other forms of such proof.

**Reply by hellmanhellman** [reply]
That sucks to hear man. I have asbergers syndrome and I tell you that because I'm picking up weird feelings like I've upset you somehow, I did not mean to.  I can't literally can't empathetize with your feelings of "best family ever" versus orphan. My family was the best. I'm the worst. whatever: No! I've never implied identity verification might be a solution. What you you think anonymous means? You're speaking arrogantly man, it's not like you. What happened to experiment? I'm provably at fault but if I'll not you better tell me. 


What I am, simply, suggesting, is this:
Axiom 1: Money is energy
Axiom 2: A domain where humans dwell and error of speech is actively protected, encouraged, and corrected can generate literally all ideas in the cosmos 
Axiom 3: Actions mean everything, speech is nothing.
Axiom 4: Individuals interacting in a free market (stacker news) can have their reputation be their price. You wanna tell her she said something  cool? Well she's famous so it'll cost you 1ksats. 

I know but clout and friend cloud or whatever fucked it up. But humans can be anonymous, non-fungible (Ugh) assets on a digital market where we coexist and share our ideas. Under ZERO top or bottom pressure. Trustless.

Action is sending LND to someone else. No one can take away that suppert from the speaker. It gives them value in my eyes. And everyone else's. 


Whats with all the logistic bounds on tipping? It's like you want you incentice bots? It should be exponentially more expensive to tip more?? If course not. Linear. 

Do you hear any of this, am I just ranting..

**Reply by huumn** [reply]
> No! I've never implied identity verification might be a solution. What you you think anonymous means?

I'm sorry for polarizing your implication. I know that's not what you meant. I was polarizing it for brevity's sake ... because the more anonymous an application the more proof of humanness it requires.

**Reply by hellmanhellman** [reply]
That's an interesting point. And we need humans for truth right? But isn't that exactly what lightning solves? If all ranking is based on and proportional to actual money paid.. Well it might be as close to anonymous honest interaction possible. I'll think more on it. Sorry for flying off the handle, i was a bit overtired :)

**Reply by huumn** [reply]
No worries dude. It happens to us all!
